src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmCenGridSpec.scala

```diff
@@ -130,48 +120,37 @@ class DBFSAlgorithmCenGridSpec
         )
 
       val key =
-        ScheduleLock.singleKey(TSpawner, scheduler.ref.toTyped, INIT_SIM_TICK)
-      scheduler.expectMsgType[ScheduleActivation] // lock activation scheduled
+        ScheduleLock.singleKey(TSpawner, scheduler.ref, INIT_SIM_TICK)
+      scheduler
+        .expectMessageType[ScheduleActivation] // lock activation scheduled
 
-      centerGridAgent ! GridAgent.Create(
+      centerGridAgent ! CreateGridAgent(
         gridAgentInitData,
         key,
       )
-      scheduler.expectMsg(
-        ScheduleActivation(centerGridAgent.toTyped, INIT_SIM_TICK, Some(key))
-      )
 
-      scheduler.send(centerGridAgent, Activation(INIT_SIM_TICK))
-      scheduler.expectMsg(
-        Completion(
-          centerGridAgent.toTyped,
-          Some(3600),
-        )
-      )
+      val msg = scheduler.expectMessageType[ScheduleActivation]
+      msg shouldBe ScheduleActivation(msg.actor, INIT_SIM_TICK, Some(key))
 
+      centerGridAgent ! WrappedActivation(Activation(INIT_SIM_TICK))
+      val completionMessage = scheduler.expectMessageType[Completion]
+      completionMessage shouldBe Completion(msg.actor, Some(3600))
     }
 
-    s"go to $SimulateGrid when it receives an activity start trigger" in {
+    s"go to SimulateGrid when it receives an activity start trigger" in {
 
-      scheduler.send(
-        centerGridAgent,
-        Activation(3600),
-      )
+      centerGridAgent ! WrappedActivation(Activation(3600))
 
-      scheduler.expectMsg(
-        Completion(
-          centerGridAgent.toTyped,
-          Some(3600),
-        )
-      )
+      val msg = scheduler.expectMessageType[Completion]
+      msg shouldBe Completion(msg.actor, Some(3600))
```

Comment:
This can be abbreviated by `scheduler.expectMessage(Completion(msg.actor, Some(3600))`. Same everywhere else.